### ISSUE TITLE
Include rule name when reporting a violation

### ISSUE THREAD

**Issue by eps1lon** [stylelint/stylelint#5112]
> What is the problem you're trying to solve?

I'm trying to understand a violation (false positive or bug?):

```
docs/types/react-docgen.d.ts
 6:11  ✖  Export 'ASTNode' is not defined (6:11)
```

Update:
It seems like the rule is `undefined`. If I use the verbose formatter it prints

```
docs/types/react-docgen.d.ts
 7:11  ✖  Export 'ASTNode' is not defined (7:11)

1 source checked
 /home/eps1lon/Development/projects/mui/fork/docs/types/react-docgen.d.ts

1 problem found
 severity level "error": 1
  undefined: 1
```

The the full text is actually `Export 'ASTNode' is not defined (7:11) (undefined)` though I don't know why the  `(undefined)` would not be printed in the console.

> What solution would you like to see?

When reporting a violation, include the rule name (like eslint does).

**Comment by m-allanson** [comment]
This is maybe stylelint trying to parse a `.d.ts` file and failing because there's nothing it can parse in that file. You might need to exclude `.d.ts` files from being passed to stylelint?

There's not enough info here to diagnose further so I'm going to close this issue.

I encourage you to open a new issue that follows the issue template. In particular, include reproduction steps and then hopefully we can work this out.

**Comment by eps1lon** [comment]
>I encourage you to open a new issue that follows the issue template. In particular, include reproduction steps and then hopefully we can work this out.

I was following the issue template for a feature request. What issue template are you referring to?

**Comment by hudochenkov** [comment]
@eps1lon bug template.

Stylelint does show lint rule in result just like ESLint.

**Comment by eps1lon** [comment]
> Stylelint does show lint rule in result just like ESLint.

I just showed you an example where it doesn't. 

But it's clear to me that you're not interested in this. The number of open issues should've made this clear. Sorry for wasting your time.

**Comment by alexander-akait** [comment]
@eps1lon I think that you just described the problem badly, just fix it, if you provide screenshot(s) that you expected - attach them

**Comment by m-allanson** [comment]
Hey @eps1lon, I apologise. This sounded like a bug report and I didn't realise you were using the feature template.

Stylelint _should_ include the rule name when reporting a violation. You have found a case where it doesn't, which indicates there may be a bug.

If you'd like to open a bug report with the steps to recreate your error, I can investigate.

**Comment by jeddy3** [comment]
Yes, it just looks like crossed wires. Let's all assume good intent if we're not already. I'd like to take this opportunity to thank all the stylelint contributors who give their time freely to triage issues. With 200+ triaged issues and over 2000 closed ones, you're all doing a fantastic job! @eps1lon I'd like to thank you too, as a fellow open source contributor, for your work on Yarn and Testing Library, which I two tools I use a lot.

> Stylelint should include the rule name when reporting a violation. You have found a case where it doesn't, which indicates there may be a bug. If you'd like to open a bug report with the steps to recreate your error, I can investigate.

It does sound like a bug. @eps1lon In the new bug report, would you mind including the snippet of code causing the error? This will help us track down the source of the problem.
